prototype2/gdgem/NMXConfig.cpp

```diff
@@ -114,38 +120,34 @@ NMXConfig::NMXConfig(std::string configfile, std::string calibrationfile) {
     clusterer_y.max_strip_gap = cy["max_strip_gap"].get<unsigned int>();
     clusterer_y.max_time_gap = cy["max_time_gap"].get<double>();
 
+    if(root.count("matcher")) {
+      matcher_name = root["matcher"].get<std::string>();
+    }
     matcher_max_delta_time = root["matcher_max_delta_time"].get<double>();
 
-    if (root.count("analyzer") && (root["analyzer"] == "MG"))
-    {
-      auto MGA = std::make_shared<MGAnalyzer>();
-      MGA->weighted(root["analyze_weighted"].get<bool>());
-      geometry.nx(MGA->geometry().x_range());
-      geometry.ny(MGA->geometry().y_range());
-      geometry.nz(MGA->geometry().z_range());
-      geometry.np(1);
-
-      analyzer_ = MGA;
+    if (root.count("analyzer")) {
```

Comment:
Are you changing the json format? If so please update the json files in DM/data/EFU_reference/gdgem 
to reflect this. If not the integration tests will fail.